src/tiledb/cloud/dag/dag.py

```diff
@@ -219,6 +220,16 @@ def _check_resources_and_mode(self) -> None:
                     )
             elif not resources_set:
                 self._resource_class = "standard"
+            # Validate the memory string because it's easy to forget the unit
+            # suffix and request "2" instead of "2Gi". Anything less than 10 Mb
+            # (written in bytes) was probably a user error.
+            if resources_set and "memory" in self._resources:
+                pattern = re.compile("^[0-9]{1,7}$")
+                if pattern.match(self._resources["memory"]) is not None:
```

Comment:
this can be

```python
if re.match(r"...", self._resources["memory"]):
  raise ...
```

just using the `re.match` function rather than compiling an intermediate expression, and using the fact that any regular expression match is truthy.

Reply:
Much nicer, thanks!